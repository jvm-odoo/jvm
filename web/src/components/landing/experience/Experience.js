import React from 'react'

import {
  ExperienceContainer,
  ExperienceImage,
  ExperienceInfo,
  ExperienceTitle,
  ExperienceDetails,
} from '../'

export const Experience = ({ image, place, from, to, title, margin }) => {
  return (
    <ExperienceContainer margin={margin}>
      <ExperienceImage src={image} alt={place} />
      <ExperienceDetails>
        <ExperienceInfo place={place} from={from} to={to} />
        <ExperienceTitle>{title}</ExperienceTitle>
      </ExperienceDetails>
    </ExperienceContainer>
  )
}