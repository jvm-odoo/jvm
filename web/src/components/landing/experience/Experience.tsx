import React from 'react'

import {
    ExperienceContainer,
    ExperienceImage,
    ExperienceInfo,
    ExperienceTitle,
    ExperienceDetails,
} from '../'

export interface IExperience {
    id: string,
    image: string,
    place: string,
    dateFrom: string,
    dateTo: string,
    title: string,
    margin: boolean,
    role: string,
}

export const Experience = ({ image, place, dateFrom, dateTo, title }: Partial<IExperience>) => {
    return (
        <ExperienceContainer>
            <ExperienceImage src={image} alt={place} />
            <ExperienceDetails>
                <ExperienceInfo place={place} dateFrom={dateFrom} dateTo={dateTo} />
                <ExperienceTitle>{title}</ExperienceTitle>
            </ExperienceDetails>
        </ExperienceContainer>
    )
}