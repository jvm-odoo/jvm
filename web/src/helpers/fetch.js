export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
