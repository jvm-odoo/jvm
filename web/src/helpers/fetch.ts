export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}