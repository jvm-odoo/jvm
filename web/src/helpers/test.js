import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
