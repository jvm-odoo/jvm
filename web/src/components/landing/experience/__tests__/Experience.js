import React from 'react'

import { Experience } from '../Experience'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<Experience />))
