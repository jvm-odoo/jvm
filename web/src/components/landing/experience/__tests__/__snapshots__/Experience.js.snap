// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.img />
  <styled.div>
    <ExperienceInfo />
    <styled.h3 />
  </styled.div>
</styled.div>
`;
